internal/auth/token_exchange.go

```diff
@@ -44,8 +42,29 @@ func FetchGrantToken(n8nURI, authToken string) (string, error) {
 
 	var tokenResp grantTokenResponse
 	if err := json.NewDecoder(resp.Body).Decode(&tokenResp); err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to decode grant token response: %w", err)
 	}
 
 	return tokenResp.Data.Token, nil
 }
+
+// FetchGrantToken exchanges the launcher's auth token for a less privileged
+// grant token from the n8n main instance. In case the n8n main instance is
+// temporarily unavailable, This exchange is retried a limited number of times.
```

Comment:
```suggestion
// FetchGrantToken exchanges the launcher's auth token for a single use
// grant token from the n8n main instance. In case the n8n main instance is
// temporarily unavailable, this exchange is retried a limited number of times.
```